presto-docs/src/main/sphinx/develop/delete-and-update.rst

```diff
@@ -0,0 +1,178 @@
+
+``DELETE`` and ``UPDATE`` have a similar flow:
+
+* For each split, the connector will create an ``UpdatablePageSource`` instance, layered over the
+  connector's ``ConnectorPageSource``, to read pages on behalf of the Presto engine, and to
+  write deletions and/or updates to the underlying data store.
+* The connector's ``UpdatablePageSource.getNextPage()`` implementation fetches the next page
+  from the underlying ``ConnectorPageSource``, optionally reformats the page, and returns it
+  to the Presto engine.
+* The Presto engine performs filtering and projection on the page read, producing a page of filtered,
+  projected results.
+* The Presto engine passes that filtered, projected page of results to the connector's
+  ``UpdatablePageSource`` ``deleteRows()`` or ``updateRows()`` method.  Those methods persist
+  the deletions or updates in the underlying data store.
+* When all the pages for a specific split have been processed, the Presto engine calls
+  ``UpdatablePageSource.finish()``, which returns a ``Collection<Slice>`` of "fragments"
+  representing connector-specific information about the rows processed by the calls to
+  ``deleteRows`` or ``updateRows``.
+* When all pages for all splits have been processed, the Presto engine calls ``ConnectorMetadata.finishDelete()`` or
+  ``finishUpdate``, passing a collection containing all the "fragments" from all the splits.  The connector
+  does what is required to finalize the operation, for example, committing the transaction.
+
+The rowId Column Abstraction
+============================
+
+The Presto engine and connectors use a "rowId" column handle abstraction to agree on the identities of rows
+to be updated or deleted.  The rowId column handle is opaque to the Presto engine.  Depending on the connector,
+the rowId column handle abstraction could represent several physical columns.
+
+The rowId Column for ``DELETE``
+-------------------------------
+
+The Presto engine identifies the rows to be deleted using a connector-specific
+rowId column handle, returned by the connector's ``ConnectorMetadata.getDeleteRowIdColumnHandle()``
+method, whose full signature is::
+
+    ColumnHandle getDeleteRowIdColumnHandle(
+        ConnectorSession session,
+        ConnectorTableHandle tableHandle)
+
+The rowId Column for ``UPDATE``
+-------------------------------
+
+The Presto engine identifies rows to be updated using a connector-specific rowId column handle,
+returned by the connector's ``ConnectorMetadata.getUpdateRowIdColumnHandle()``
+method.  In addition to the columns that identify the row, for ``UPDATE`` the rowId column will contain
+any columns that the connector requires in order to perform the ``UPDATE`` operation.
+
+UpdatablePageSource API
+=======================
+
+As mentioned above, to support ``DELETE`` or ``UPDATE``, the connector must define a subclass of
+``UpdatablePageSource``, layered over the connector's usual ``ConnectorPageSource``.  The interesting methods are:
```

Comment:
```suggestion
``UpdatablePageSource``, layered over the connector's ``ConnectorPageSource``.  The interesting methods are:
```
As above, suggest deleting "usual" - unless there is a meaning to its presence that I do not understand. If "usual" is meaningful, can you help me to understand why?